server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -78,11 +79,27 @@ public void execute(SearchContext context) {
 
         if (context.docIdsToLoadSize() == 0) {
             // no individual hits to process, so we shortcut
-            context.fetchResult().hits(new SearchHits(new SearchHit[0], context.queryResult().getTotalHits(),
-                context.queryResult().getMaxScore()));
+            SearchHits hits = new SearchHits(new SearchHit[0], context.queryResult().getTotalHits(), context.queryResult().getMaxScore());
+            context.fetchResult().shardResult(hits, null);
             return;
         }
 
+        Profiler profiler = context.getProfilers() == null ? Profiler.NOOP : context.getProfilers().getFetchProfiler();
```

Comment:
I wasn't super happy with how invasive the change was here. I'm happy that the sub-phases for the most part don't have to change which is good, but this bit is bigger than I'd like. I'd love any suggestions if anyone has any.